pkgs/development/libraries/libucl/default.nix

```
@@ -0,0 +1,57 @@
+{ stdenv
+, fetchFromGitHub
+, pkgconfig
+, autoreconfHook
+, curl
+, lua
+, openssl
+, features ? {
+    urls = false;
+    # Upstream enables regex by default
+    regex = true;
+    # Signature support is broken with openssl 1.1.1: https://github.com/vstakhov/libucl/issues/203
+    signatures = false;
+    lua = false;
+    utils = false;
+  }
+}:
+
+let
+  featureDeps = {
+    urls = [ curl ];
+    signatures = [ openssl ];
+    lua = [ lua ];
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "libucl";
+  version = "0.8.1";
+
+  nativeBuildInputs = [ pkgconfig autoreconfHook ] ++
+  stdenv.lib.concatLists (
+    stdenv.lib.mapAttrsToList (
+      name: value:
+      stdenv.lib.optionals value (featureDeps."${name}" or [])
```

If you use `or`, then `lib.optionals` is an overkill. Plus, perhaps it'll be nicer to rename that function's arguments:

```suggestion
      feat: enable:
      if enable then featureDeps."${feat}" else []
```

---

This won't work, as `featureDeps` doesn't map 1-1 with `feature` attrSet. I added the `or` in order to provide a default value when `feat` is not found in `featureDeps`.

---

I've renamed the arguments though.